Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/LocationUpdateService.java

```diff
@@ -167,7 +185,52 @@ public int onStartCommand(Intent intent, int flags, int startId) {
         return START_STICKY;
     }
 
-    private void updateDeviceDetails() {
+    private void startDistanceCalculation() {
+        LocationCallback locCallback = getDistanceCalCallback();
+        LocationRequest locReq = createDistanceCalculation(2000, 0, 2000);
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            return;
+        }
+        fusedLocClientForDistanceCal.requestLocationUpdates(locReq, locCallback, Looper.getMainLooper());
+    }
+
+    private LocationCallback getDistanceCalCallback() {
+        calDistanceCallback = new LocationCallback() {
+            @Override
+            public void onLocationResult(@NonNull LocationResult locationResult) {
+                if (ActivityCompat.checkSelfPermission(getApplicationContext(), Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(getApplicationContext(), Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+                    return;
+                }
+                fusedLocClientForDistanceCal.getCurrentLocation(Priority.PRIORITY_HIGH_ACCURACY, cancellationTokenSource.getToken())
+                        .addOnSuccessListener(location -> {
+                            if(location!=null) {
+                                isDistanceCalulation = true;
+                                int distanceBtLatLng = 0, distanceBtLatLngWithAcc = 0;
+                                String accHolder = getValueFromStorage("ACCURACY_THRESHOLD");
+                                String accuracyThreshold = accHolder == null ? "100.0" : accHolder;
+                                if (prevLocation != null && prevAccLocation != null) {
+                                    distanceBtLatLng = (int) location.distanceTo(prevLocation);
+                                    finalDistance+=distanceBtLatLng;
```

Comment:
is finalDistance dependent on prevAccLocation value as if it is null it won't be upated